src/Native/System.Native/pal_io.cpp

```diff
@@ -383,13 +383,40 @@ extern "C" int32_t SystemNative_Pipe(int32_t pipeFds[2], int32_t flags)
 
     int32_t result;
 #if HAVE_PIPE2
+    // If pipe2 is available, use it.  This will handle O_CLOEXEC if it was set.
     while (CheckInterrupted(result = pipe2(pipeFds, flags)));
 #else
-    while (CheckInterrupted(result = pipe(pipeFds)));         // CLOEXEC intentionally ignored on platforms without pipe2.
+    // Otherwise, use pipe.
+    while (CheckInterrupted(result = pipe(pipeFds)));
+    
+    // Then, if O_CLOEXEC was specified, use fcntl to configure the file descriptors appropriately.
+    if ((flags & O_CLOEXEC) != 0 && result == 0)
+    {
+        while (CheckInterrupted(result = fcntl(pipeFds[0], F_SETFD, FD_CLOEXEC)));
```

Comment:
I forgot about the PAL types could be an issue. Maybe pulling all 3 calls into a static "private" method would get around that. But I don't really feel strongly.

If you want to do that tiny refactoring, go for it. If you'd rather not, I won't block you. :)